File: Level_2/가장_큰_수.js
function solution(numbers) {
  let answer = '';
  let per = [];

  for (let i = 0; i < numbers.length; i++) {
    numbers[i] = numbers[i].toString();
  }

  let permutation = function (i, numbers) {
    if (i === numbers.length) {
      per.push(numbers.slice());
      return;
    }

    for (let j = i; j < numbers.length; j++) {
      [numbers[i], numbers[j]] = [numbers[j], numbers[i]];
      permutation(i + 1, numbers);
      [numbers[i], numbers[j]] = [numbers[j], numbers[i]];
    }
  };

  permutation(0, numbers);

  for (let i = 0; i < per.length; i++) {
    let temp = '';
    for (let j = 0; j < per[i].length; j++) {
      temp += per[i][j];
    }
    per[i] = parseInt(temp);
  }

  answer = per[0];

  for (let i = 1; i < per.length; i++) {
    if (answer < per[i]) answer = per[i];
  }

  answer = answer.toString();
  return answer;
}

let numbers = [6, 10, 2];
console.log(solution(numbers));
let numbers_1 = [3, 30, 34, 5, 9];
console.log(solution(numbers_1));

File: Level_2/가장_큰_수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string solution(vector<int> numbers) {
    string answer = "";

    vector<string> nums;
    for (int i = 0; i < numbers.size(); i++) {
        nums.push_back(to_string(numbers[i]));
    }

    sort(nums.begin(), nums.end(), [](const string& a, const string& b) {
        return a + b > b + a;
    });

    for (int i = 0; i < nums.size(); i++) {
        answer += nums[i];
    }

    if (answer[0] == '0') {
        answer = "0";
    }

    return answer;
}
